fastjet/internal/delaunay/triangle.go

```diff
@@ -0,0 +1,201 @@
+package delaunay
+
+import (
+	"fmt"
+	"math"
+)
+
+type Triangle struct {
+	children []*Triangle
+	A        *Point
+	B        *Point
+	C        *Point
+}
+
+func NewTriangle(a *Point, b *Point, c *Point) *Triangle {
+	// make it counter clockwise
+	if !cClockwise(a, b, c) {
+		temp := b
+		b = c
+		c = temp
+	}
+	t := &Triangle{
+		A: a,
+		B: b,
+		C: c,
+	}
+	t.children = make([]*Triangle, 0, 10)
+	return t
+}
+
+func cClockwise(a *Point, b *Point, c *Point) bool {
+	return (b.y-a.y)*(c.x-b.x)-(c.y-b.y)*(b.x-a.x) < 0
+}
+
+// checking if the point is in the triangle and if it is on an edge
+// using barycentric coordinates
+func (t *Triangle) inTriangle(p *Point) (bool, bool) {
+	barcen1 := ((t.B.y-t.C.y)*(p.x-t.C.x) + (t.C.x-t.B.x)*(p.y-t.C.y)) / det(t)
+	barcen2 := ((t.C.y-t.A.y)*(p.x-t.C.x) + (t.A.x-t.C.x)*(p.y-t.C.y)) / det(t)
+	barcen3 := 1 - barcen1 - barcen2
+
+	// inside triangle
+	if barcen1 > 0 && barcen1 < 1 && barcen2 > 0 && barcen2 < 1 && barcen3 > 0 && barcen3 < 1 {
+		return true, false
+	}
+	// outside triangle
+	if barcen1 < 0 || barcen2 < 0 || barcen3 < 0 {
+		return false, false
+	}
+	// otherwise on the edge
+	return true, true
+}
+
+func det(t *Triangle) float64 {
+	return (t.B.y-t.C.y)*(t.A.x-t.C.x) + (t.C.x-t.B.x)*(t.A.y-t.C.y)
+}
+
+// return true if the point is inside the circumcircle of the triangle
+func (t *Triangle) inCCircle(p *Point) bool {
+	x, y := t.centCCircle()
+	// r is the radius of the circumcircle
+	r := math.Sqrt((x-t.A.x)*(x-t.A.x) + (y-t.A.y)*(y-t.A.y))
+	// point is in circumcircle if distance to center is less than radius
+	return math.Sqrt((p.x-x)*(p.x-x)+(p.y-y)*(p.y-y)) < r
```

Comment:
Since it is very expensive does it make sense to keep the distances squared internally since it is used only for comparisons? And add a method to get the distance for the user, where the square root is taken.